### PR TITLE
Merge master into json-output

### PR DESCRIPTION
**Pull request by celic**

### CODE REVIEW

**Comment by celic** [comment]
@joshuaehill I'm having a hard time reading these comments with the lines they are referencing. Could these be moved to issues so I can address them?

**Comment by joshuaehill** [comment]
I didn't know that my comments ended up being rendered like this; I made the comments inline to a commit diff, and the comments make somewhat more sense in that context (with the comment appearing directly under the code that I am commenting on). I'll write up a proper issue or two. Sorry for any confusion!

File: cpp/non_iid_main.cpp [hunk 1]
```diff
@@ -217,105 +217,101 @@ int main(int argc, char* argv[]) {
...
         if (verbose == 1) printf("\tMost Common Value Estimate (bit string) = %f / 1 bit(s)\n", ret_min_entropy);
         H_bitstring = min(ret_min_entropy, H_bitstring);
+        tc631.ret_min_entropy = ret_min_entropy;
     }
 
     if (initial_entropy) {
         ret_min_entropy = most_common(data.symbols, data.len, data.alph_size, verbose, "Literal", tc631);
         if (verbose == 1) printf("\tMost Common Value Estimate = %f / %d bit(s)\n", ret_min_entropy, data.word_size);
         H_original = min(ret_min_entropy, H_original);
+        tc631.ret_min_entropy = ret_min_entropy;
```
Comment (joshuaehill) [inline cpp/non_iid_main.cpp:232]: In cases where a test performs both literal and bitstring estimates, shouldn't both be saved? In this instance, this code saves only the literal estimate.

File: cpp/non_iid_main.cpp [hunk 2]
```diff
@@ -217,105 +217,101 @@ int main(int argc, char* argv[]) {
...
         if (verbose == 1) printf("\tCollision Test Estimate (bit string) = %f / 1 bit(s)\n", ret_min_entropy);
         H_bitstring = min(ret_min_entropy, H_bitstring);
+        tc632.ret_min_entropy = ret_min_entropy;
     }
 
     if (initial_entropy && (data.alph_size == 2)) {
         ret_min_entropy = collision_test(data.symbols, data.len, verbose, "Literal");
         if (verbose == 1) printf("\tCollision Test Estimate = %f / 1 bit(s)\n", ret_min_entropy);
         H_original = min(ret_min_entropy, H_original);
+        tc632.ret_min_entropy = ret_min_entropy;
```
Comment (joshuaehill) [inline cpp/non_iid_main.cpp:254]: Similarly, if both literal and bitstring assessments are conducted, only the literal assessment is saved.

File: cpp/non_iid_main.cpp [hunk 3]
```diff
@@ -217,105 +217,101 @@ int main(int argc, char* argv[]) {
...
         if (verbose == 1) printf("\tMarkov Test Estimate (bit string) = %f / 1 bit(s)\n", ret_min_entropy);
         H_bitstring = min(ret_min_entropy, H_bitstring);
+        tc633.ret_min_entropy = ret_min_entropy;
     }
 
     if (initial_entropy && (data.alph_size == 2)) {
         ret_min_entropy = markov_test(data.symbols, data.len, verbose, "Literal");
         if (verbose == 1) printf("\tMarkov Test Estimate = %f / 1 bit(s)\n", ret_min_entropy);
         H_original = min(ret_min_entropy, H_original);
+        tc633.ret_min_entropy = ret_min_entropy;
```
Comment (joshuaehill) [inline cpp/non_iid_main.cpp:274]: Similarly, if both literal and bitstring assessments are conducted, only the literal assessment is saved.

File: cpp/non_iid_main.cpp [hunk 4]
```diff
@@ -217,105 +217,101 @@ int main(int argc, char* argv[]) {
...
         }
     }
 
     if (initial_entropy && (data.alph_size == 2)) {
         ret_min_entropy = compression_test(data.symbols, data.len, verbose, "Literal");
         if (ret_min_entropy >= 0) {
             if (verbose == 1) printf("\tCompression Test Estimate = %f / 1 bit(s)\n", ret_min_entropy);
-            H_original = min(ret_min_entropy, H_original);    
+            H_original = min(ret_min_entropy, H_original);
+            tc634.ret_min_entropy = ret_min_entropy;
```
Comment (joshuaehill) [inline cpp/non_iid_main.cpp:297]: Similarly, if both literal and bitstring assessments are conducted, only the literal assessment is saved.

File: cpp/non_iid_main.cpp [hunk 5]
```diff
@@ -324,51 +320,42 @@ int main(int argc, char* argv[]) {
         if (t_tuple_res >= 0.0) {
             if (verbose == 1) printf("\tT-Tuple Test Estimate = %f / %d bit(s)\n", t_tuple_res, data.word_size);
             H_original = min(t_tuple_res, H_original);
+            tc635.t_tuple_res = t_tuple_res;
```
Comment (joshuaehill) [inline cpp/non_iid_main.cpp:323]: Similarly, if both literal and bitstring assessments are conducted, only the literal assessment is saved.

File: cpp/non_iid_main.cpp [hunk 6]
```diff
@@ -377,27 +364,22 @@ int main(int argc, char* argv[]) {
         if (ret_min_entropy >= 0) {
             if (verbose == 1) printf("\tMulti Most Common in Window (MultiMCW) Prediction Test Estimate = %f / %d bit(s)\n", ret_min_entropy, data.word_size);
             H_original = min(ret_min_entropy, H_original);
+            tc637.ret_min_entropy = ret_min_entropy;
```
Comment (joshuaehill) [inline cpp/non_iid_main.cpp:367]: Similarly, if both literal and bitstring assessments are conducted, only the literal assessment is saved.

File: cpp/non_iid_main.cpp [hunk 7]
```diff
@@ -406,27 +388,22 @@ int main(int argc, char* argv[]) {
         if (ret_min_entropy >= 0) {
             if (verbose == 1) printf("\tLag Prediction Test Estimate = %f / %d bit(s)\n", ret_min_entropy, data.word_size);
             H_original = min(ret_min_entropy, H_original);
+            tc638.ret_min_entropy = ret_min_entropy;
```
Comment (joshuaehill) [inline cpp/non_iid_main.cpp:391]: Similarly, if both literal and bitstring assessments are conducted, only the literal assessment is saved.

File: cpp/non_iid_main.cpp [hunk 8]
```diff
@@ -435,27 +412,22 @@ int main(int argc, char* argv[]) {
         if (ret_min_entropy >= 0) {
             if (verbose == 1) printf("\tMulti Markov Model with Counting (MultiMMC) Prediction Test Estimate = %f / %d bit(s)\n", ret_min_entropy, data.word_size);
             H_original = min(ret_min_entropy, H_original);
+            tc639.ret_min_entropy = ret_min_entropy;
```
Comment (joshuaehill) [inline cpp/non_iid_main.cpp:415]: Similarly, if both literal and bitstring assessments are conducted, only the literal assessment is saved.

File: cpp/non_iid_main.cpp [hunk 9]
```diff
@@ -464,22 +436,24 @@ int main(int argc, char* argv[]) {
         if (ret_min_entropy >= 0) {
             if (verbose == 1) printf("\tLZ78Y Prediction Test Estimate = %f / %d bit(s)\n", ret_min_entropy, data.word_size);
             H_original = min(ret_min_entropy, H_original);
+            tc6310.ret_min_entropy = ret_min_entropy;
```
Comment (joshuaehill) [inline cpp/non_iid_main.cpp:439]: Similarly, if both literal and bitstring assessments are conducted, only the literal assessment is saved.

**Comment by joshuaehill** [comment]
@celic I think that this PR can be closed.

**Pull request closed without merging**